lib/movie_widgets/movie_genres.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+import 'package:movie_db_app/repositories_i/repository_i.dart';
+import '/data_classes/genre.dart';
+import '/constants/constants.dart';
+
+import '/repositories/genres_local_repository.dart';
+
+class MovieGenres extends StatefulWidget {
+  final List<int> movieGenres;
+
+  const MovieGenres({
+    super.key,
+    required this.movieGenres,
+  });
+
+  List<String> linkGenres(List<Genre> genresMap) {
+    return genresMap
+        .where((genre) => movieGenres.contains(genre.id))
+        .map((genre) => genre.name)
+        .toList();
+  }
+
+  @override
+  State<StatefulWidget> createState() => _MovieGenresState();
+}
+
+class _MovieGenresState extends State<MovieGenres> {
+  final IRepository genreRepository = GenreLocalRepository();
+  late Future<List<Genre>> genresList;
+
+  @override
+  initState() {
+    super.initState();
+    genresList = genreRepository.loadData() as Future<List<Genre>>;
```

Comment:
I'd move this cast above:
`final IRepository<List<Genre>> genreRepository = GenreLocalRepository();`

Reply:
fixed